lightning/src/chain/channelmonitor.rs

```diff
@@ -276,7 +276,7 @@ impl HolderSignedTx {
 /// justice or 2nd-stage preimage/timeout transactions.
 #[derive(Clone, PartialEq, Eq)]
 struct CounterpartyCommitmentParameters {
-	counterparty_delayed_payment_base_key: PublicKey,
+	counterparty_delayed_payment_base_key: DelayedPaymentBasepoint,
```

Comment:
BasePoint and BaseSecret would be 1 to 1 mapping to PublicKey and SecretKey, so what is the value to do it. Instead we should create types that separate between **any key** used in the channel: `PaymentBasepoint` vs `DelayedPaymentBasepoint` vs `HtlcBasepoint`. Then the value would be to help a dev to not mix up the keys and find them faster.

Reply:
Yes, it's one of the confusions and having `Basepoint` as a separate type would help with that. 

I was just thinking in a bit wider scope, that it might be useful to engage Rust type system to check if the key type used is correct and rust-analyzer to show what key is being returned, etc, also having an option to search for a specific key in the codebase.